t/lib/t/MusicBrainz/Server/Controller/WS/2/DefaultFormat.pm

```diff
@@ -0,0 +1,91 @@
+package t::MusicBrainz::Server::Controller::WS::2::DefaultFormat;
+
+use JSON qw( decode_json encode_json );
+use Test::Deep qw( cmp_deeply );
+use Test::JSON import => [qw( is_valid_json is_json )];
+use Test::More;
+use Test::Routine;
+use Test::XML::SemanticCompare;
+
+with 't::Mechanize', 't::Context';
+
+test 'MBS-9258: deduction of fmt from Accept header' => sub {
```

Comment:
You're right - it tests the code that was changed for MBS-9258, but not the MBS-9258 use case.
I'll look at adding JSON cases (with and without explicit fmt) to that indexed search test.